lib/cpp/src/thrift/transport/TServerSocket.cpp

```diff
@@ -60,7 +60,7 @@
 #endif // _WIN32
 #endif
 
-#if defined(_WIN32) && (_WIN32_WINNT < 0x0600)
+#if defined(_WIN32) && (_WIN32_WINNT < 0x0600) && !defined(AI_ADDRCONFIG)
   #define AI_ADDRCONFIG 0x0400
```

Comment:
You didn't break this, but this kind of thing is exactly what the platform socket header is for.  Defining constants that don't belong to us often leads to trouble.  Instead, we should be defining something like THRIFT_AI_ADDRCONFIG that can either be the real AI_ADDRCONFIG or 0x0400.